MsixCore/msixmgr/msixmgr.cpp

```diff
@@ -347,32 +552,63 @@ int main(int argc, char * argv[])
                     std::wcout << "Failed to create temp directory " << tempDirPathString << std::endl;
                     std::wcout << "This may occur when the directory path already exists. Please try again."  << std::endl;
                     std::wcout << std::endl;
+
+                    // Telemetry : Workflow Log
+                    QueryPerformanceCounter(&msixMgrLoad_EndCounter);
+                    workflowElapsedTime = msixmgr::CalcWorkflowElapsedTime(msixMgrLoad_StartCounter, msixMgrLoad_EndCounter, msixMgrLoad_Frequency);
+                    msixmgr::TraceLogWorkflow(workflowId.c_str(), cli.GetOperationTypeAsString().c_str(), false, workflowElapsedTime, errorCode.c_str(), errorDesc.c_str());
+
                     return E_UNEXPECTED;
                 }
                 if (createDirectoryErrorCode.value() != 0)
                 {
+                    errorCode = msixmgr::GetErrorCodeFromHRESULT(E_UNEXPECTED);
+                    errorDesc = L"Failed to create temp directory for CIM flow with Error Message: " + utf8_to_utf16(createDirectoryErrorCode.message()) + L". Need to try again. HRESULT " + errorCode + L".";
+
                     // Again, we expect that the creation of the temp directory will fail very rarely. Output the exception
                     // and have the user try again.
                     std::wcout << std::endl;
                     std::wcout << "Creation of temp directory " << tempDirPathString << " failed with error: " << createDirectoryErrorCode.value() << std::endl;
                     std::cout << "Error message: " << createDirectoryErrorCode.message() << std::endl;
                     std::wcout << "Please try again." << std::endl;
                     std::wcout << std::endl;
+
+                    // Telemetry : Workflow Log
+                    QueryPerformanceCounter(&msixMgrLoad_EndCounter);
+                    workflowElapsedTime = msixmgr::CalcWorkflowElapsedTime(msixMgrLoad_StartCounter, msixMgrLoad_EndCounter, msixMgrLoad_Frequency);
+                    msixmgr::TraceLogWorkflow(workflowId.c_str(), cli.GetOperationTypeAsString().c_str(), false, workflowElapsedTime, errorCode.c_str(), errorDesc.c_str());
+
                     return E_UNEXPECTED;
                 }
 
-                RETURN_IF_FAILED(MsixCoreLib::Unpack(
+                HRESULT hrUnpackToTempDir = MsixCoreLib::Unpack(
                     packageSourcePath,
                     tempDirPathString,
                     cli.IsApplyACLs(),
                     cli.IsValidateSignature(),
                     skippedFiles,
                     failedPackages,
-                    failedPackagesErrors));
+                    failedPackagesErrors);
+
+                if (FAILED(hrUnpackToTempDir))
+                {
+                    errorCode = msixmgr::GetErrorCodeFromHRESULT(hrUnpackToTempDir);
+                    errorDesc = L"Failed to Unpack in Temp Directory for CIM flow. HRESULT " + errorCode + L".";

```

Comment:
Instead of "CIM flow", we can use same text as mentioned in above comment

Reply:
As discussed, all the Telemetry variables will have the same Error Message as the actual Console statements. The change for outputting the Error Message on Console from a variable will be done as part of Error Message Improvement Task that will be taken later.